### DISCUSSION TITLE
AWS Custodian Policy for Cost

### DISCUSSION THREAD

**Discussion by mmusleh75** [cloud-custodian/cloud-custodian#7369] [General]
Is there a way to create an AWS custodian policy that would notify me when an AWS resource cost goes above a threshold?

**Comment by kapilt** [comment]
it would be helpful if you could elaborate on use case. ie. first what do you mean by resource ? an individual resource, a set of resources across different resource types? sharing a common existing tag ? etc. and then what do you want in the notification.

fwiw, there's native capabilities for that in AWS Budgets. For custodian its a bit unclear as those would span many resource types, and budget forecasting gets complex when factoring in reserved & volume discounts/blended costs/savings plans .. etc.

**Comment by KVInventoR** [comment]
It's possible to use AWS Cost Explorer 
But for me it looks like as a big new part for CC deployment.

**Comment by kapilt** [comment]
the question is understanding use cases, ie. what does a cost filter mean in this context, whats would you want it to do.

**Comment by KVInventoR** [comment]
as example, it's possible to get Costs for some period for any resource in account by service name or tags.
example: 
I would like to get info how much money spend this team, or another example:  get report who spend more money in current month for s3 buckets.

**Comment by kapilt** [comment]
could you give a custodian policy example (science fiction) of what your looking, ie are you looking for cost grouped by tag per resource type, or across all account resources, or to a single resource.

**Comment by kapilt** [comment]
related #3208

**Comment by mmusleh75** [comment]
here are some of the use cases
I need to know how much did I pay over a period of time using any or none of the following filters
- resource type such as ec2, rds, s3, ... etc
- instance family
- instance size

I need to set up a custodian policy that will notify via email, or dump the data in a file if an aws resource cost me an amount of $$ more than a threshold I defined in the policy

something like this
policies:
  - name: above-threshold
    resource: ec2
    filters:
      - type: threshold
        value: 100
        period: 24 # hours
        op: gt
    actions:
      - type: notify
        ...

the intent of this policy is that if any ec2 instance cost me more than $100 within the last 24 hrs, then notify me with all instances that fit that criteria

**Comment by aakifshaikh** [comment]
Use Case:
1) I would like to identify the cost of running Cloud Custodian - meaning, we can find the cost of all resources which are tag as cloudcustodianservice. This will help us identify the cost based on the tags.
2) Identify the cost for DBs overall?
3) Identify the spike in the cost of elasticsearch (during what time in the day?)
I think we should have cost-explorer as resource type to be able to filter and take actions using custodian. I love the fact we have this in AZURE.

**Comment by alekhya114** [comment]
Any update on this use case?

**Comment by samanXgrid** [comment]
Any update on this use case. Can we achieve it using cloud custodian ?

**Comment by muhdsaad** [comment]
Is there an update on this use case?
